src/Symfony/Component/OptionsResolver/Tests/OptionsResolverTest.php

```diff
@@ -2373,4 +2373,41 @@ public function testAccessToParentOptionFromNestedNormalizerAndLazyOption()
         ];
         $this->assertSame($expectedOptions, $actualOptions);
     }
+
+    public function testResolveOptionsDefinedByOptionConfigurator()
+    {
+        $this->resolver->define('foo')
+            ->required()
+            ->deprecated()
+            ->default('bar')
+            ->allowedTypes('string')
+            ->allowedValues(['bar', 'baz']);
+
+        $this->assertTrue($this->resolver->isDeprecated('foo'));
+        $this->assertTrue($this->resolver->isRequired('foo'));
+        $this->assertSame(['foo' => 'bar'], $this->resolver->resolve());
+    }
+
+    public function testSetNormalizerOnOptionConfigurator()
+    {
+        $this->resolver->define('foo')
+            ->default('bar')
+            ->normalize(function () {
+                return 'normalized';
+            });
+
+        $this->assertEquals(['foo' => 'normalized'], $this->resolver->resolve());
+    }
+
+    public function testResolveOptionFailsIfInvalidType()
```

Comment:
I think we don't need to test this behavior again, it's already covered. We should focus on the definition only. It should be removed.